contracts/voting-verifier/src/execute.rs

```diff
@@ -92,54 +80,46 @@ pub fn verify_messages(
 
     let config = CONFIG.load(deps.storage)?;
 
-    let messages = messages
-        .into_iter()
-        .map(|message| {
-            is_message_verified(deps.as_ref(), &message).map(|verified| {
-                if verified {
-                    Verified(message)
-                } else {
-                    Pending(message)
-                }
-            })
-        })
-        .collect::<Result<Vec<VerificationStatus>, ContractError>>()?;
-
     let response = Response::new().set_data(to_binary(&VerifyMessagesResponse {
-        verification_statuses: messages
-            .iter()
-            .map(|status| match status {
-                Verified(message) => (message.cc_id.clone(), true),
-                Pending(message) => (message.cc_id.clone(), false),
-            })
-            .collect(),
+        verification_statuses: is_verified(deps.as_ref(), &messages)?,
     })?);
 
-    let pending_messages: Vec<Message> = messages
+    let messages = messages
         .into_iter()
-        .filter_map(|status| match status {
-            Pending(message) => Some(message),
-            Verified(_) => None,
+        .map(|message| verification_status(deps.as_ref(), &message).map(|status| (status, message)))
+        .collect::<Result<Vec<(VerificationStatus, Message)>, ContractError>>()?;
+
+    let msgs_to_verify: Vec<Message> = messages
+        .into_iter()
+        .filter_map(|(status, message)| match status {
+            VerificationStatus::FailedToVerify | VerificationStatus::NotVerified => Some(message),
+            VerificationStatus::InProgress | VerificationStatus::Verified => None,
         })
         .collect();
 
-    if pending_messages.is_empty() {
+    if msgs_to_verify.is_empty() {
         return Ok(response);
     }
 
-    let snapshot = take_snapshot(deps.as_ref(), &pending_messages[0].cc_id.chain)?;
+    let snapshot = take_snapshot(deps.as_ref(), &msgs_to_verify[0].cc_id.chain)?;
     let participants = snapshot.get_participants();
     let id = create_messages_poll(
         deps.storage,
         env.block.height,
         config.block_expiry,
         snapshot,
-        pending_messages.len(),
+        msgs_to_verify.len(),
     )?;
 
-    PENDING_MESSAGES.save(deps.storage, id, &pending_messages)?;
+    for (idx, message) in msgs_to_verify.iter().enumerate() {
+        POLL_MESSAGES.save(
+            deps.storage,
+            &message.hash_id(),
+            &state::PollMessage::new(message.clone(), id, idx),
+        )?;
+    }
 
-    let evm_messages = pending_messages
+    let evm_messages = msgs_to_verify
```

Comment:
Why `evm_messages`?

Reply:
oh that's right, the voter shouldn't make any mentions of specific chains